tests/scripts/thread-cert/border_routing/test_dnssd_server.py

```diff
@@ -0,0 +1,281 @@
+#!/usr/bin/env python3
```

Comment:
Lets rename this directory to `border_router` and place all border router related tests in this directory, includes:
- border routing
- DNS-SD query
- advertising proxy
- discovery proxy

Thoughts?

Reply:
We'd better resolve this in a separate PR?